src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -1,23 +1,79 @@
 package com.khutircraftubackend.product;
 
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
 @RestController
-@RequestMapping("/product")
+@RequestMapping("/products")
 @Slf4j
+@RequiredArgsConstructor
 public class ProductController {
+    private final ProductService productService;
 
+    @PostMapping("/create")
+    @PreAuthorize("hasRole('SELLER')")
+    public ResponseEntity<ProductEntity> createProduct(
+            @ModelAttribute ProductCreateRequest request,
+            @RequestParam("thumbnailImage")MultipartFile thumbnailImage,
+            @RequestParam("image") MultipartFile image) throws IOException {
+        ProductEntity newProduct = productService.createProduct(request, thumbnailImage, image);
 
+        return new ResponseEntity<>(newProduct, HttpStatus.CREATED);
+    }
+
+    @PatchMapping("/patch-product/{productId}")
+    @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
```

Comment:
proposal: 
just check if product belongs to the seller.
- pass Principal into `patchProduct` 
- `canModifyProduct` now can get `productId` and `principal` from product controller
- resolve Seller by principal in canModifyProduct
- check for equality for the whole seller if possible, not only "company Name"

Reply:
What is the purpose of these changes? We want to avoid using SecurityContextHolder in services? Is it not enough for us to check the name of the company and we need to check the whole object?